src/java.base/share/classes/sun/security/ssl/X509Authentication.java

```diff
@@ -201,6 +201,10 @@ public static SSLPossession createPossession(
     private static SSLPossession createClientPossession(
             ClientHandshakeContext chc, String[] keyTypes) {
         X509ExtendedKeyManager km = chc.sslContext.getX509KeyManager();
+        if (SSLLogger.isOn && SSLLogger.isOn("ssl")) {
+            SSLLogger.finest("X509ExtendedKeyManager being used: " +
+                    (km == null ? "null" : km.getClass().getName()));
```

Comment:
do you need to cater for null ? I thought a Dummy manager is returned in such scenarios.

Reply:
No, I did verify that it always returns Dummy Manager even when key manager is initialized as null. I have added this as a safety net. I'll remove this.